R/zchunk_LB165.ag_water_R_C_Y_GLU_irr.R

```diff
@@ -0,0 +1,407 @@
+
+    # Change L165.ag_Water_ctry_MHcrop_GLU$coef_m3kg to be the minimum of itself and
+    # the corresponding (by crop and water_type) value in L165.MaxWaterCoefs_m3kg
+    L165.MaxWaterCoefs_m3kg %>%
+      gather(water_type, value_other, Blue, Green) %>%
+      mutate(water_type = tolower(water_type)) %>%
+      right_join(L165.ag_Water_ctry_MHcrop_GLU, by = c("MH_crop", "water_type")) %>%
+      mutate(coef_m3kg = pmin(coef_m3kg, value_other)) %>%
+      select(-value_other, -FAO_crop) ->
+      L165.ag_Water_ctry_MHcrop_GLU
+
+    # Work through calculation sequence to determine blue and green water coefficients for irrigated
+    # and rainfed production for gridded crops (original lines 108-122). The sequence has 5 steps:
+    #	(1) Calculate the total biophysical (green + blue) water quantities and coefficients for each country, GLU, and crop
+    #	(2) Match in the irrigated production and calculate the blue water coef for irrigated as min(blue water quantity / irr prod, total biophysical coef)
+    #	(3) Calculate the green water coef for irrigated crops as the total biophysical coef minus the blue water coef
+    #	(4) Calculate the total volume of green water on rainfed crops, as the total green water minus green water used by irrigated crops
+    #	(5) Calculate the green water coef for rainfed crops as rainfed green water volume divided by rainfed production
+    L165.ag_Water_ctry_MHcrop_GLU %>%
+      ungroup %>%
+      select(-MH_crop, -value) %>%
+      mutate(water_type = paste0(water_type, "_m3kg")) %>%
+      spread(water_type, coef_m3kg, fill = 0.0) %>%
+      mutate(total_m3kg = blue_m3kg + green_m3kg,
+             blue_thousm3 = Prod_t * blue_m3kg,
+             green_thousm3 = Prod_t * green_m3kg) ->
+      L165.ag_Water_ctry_MHcrop_GLU
+
+    # Match in the irrigated production to calculate the blue water coef for irrigated crops (original lines 124-147)
+    L165.ag_Water_ctry_MHcrop_GLU %>%
+      left_join_error_no_match(L151.ag_irrProd_t_ctry_crop, by = c("iso", "GTAP_crop", "GLU")) %>%
+      rename(irrProd_t = irrProd) %>%
+      mutate(BlueIrr_m3kg = blue_thousm3 / irrProd_t) %>%
+      # Set the coefs to zero wherever irrigated production is zero (these couldn't have any water consumption in GCAM anyway)
+      mutate(BlueIrr_m3kg = if_else(irrProd_t == 0, 0, BlueIrr_m3kg),
+             # Cap blue water coefficients at the total biophysical quantity
+             BlueIrr_m3kg = pmin(BlueIrr_m3kg, total_m3kg),
+             # Green water coefs of irrigated crops = total biophysical coef - calculated blue water coef
+             GreenIrr_m3kg = total_m3kg - BlueIrr_m3kg,
+             # Green water quantities for rainfed crops = total green water - green water on irrigated
+             GreenIrr_thousm3 = GreenIrr_m3kg * irrProd_t,
+             GreenRfd_thousm3 = green_thousm3 - GreenIrr_thousm3) %>%
+      left_join_error_no_match(L151.ag_rfdProd_t_ctry_crop, by = c("iso", "GTAP_crop", "GLU")) %>%
```

Comment:
Comment explicitly stating that are joining in rfd crop production values to calculate the green water coefficient for rfd crops would probably be helpful to external user.